### DISCUSSION TITLE
Tool shows successful downloads but I can not find it anywhere

### DISCUSSION THREAD

**Discussion by havocart** [danielfernau/unifi-protect-video-downloader#322] [General]
For starters, I have been baby-stepping through the instructions over the past week and reading anything I can find here or on reddit or the unifi forum;  wherever you have answered questions over the past year or 2.

My Code is:

docker run --volume /downloads:/downloads unifitoolbox/protect-archiver download --address 192.168.*.*** --username .*** --password *** --start 2023-02-18T07:00:00 --end 2023-02-19T12:00:00  /downloads

The response is a variation of:

"Downloading footage for camera 'G4 Pro' (**************03ee)
Created path /downloads/2023/02/19/G4 Pro (03ee)
Downloading video for time range 2023-02-19 07:00:00 - 2023-02-19 07:59:59 to /downloads/2023/02/19/G4 Pro (03ee)/G4 Pro (03ee) - 2023-02-19 - 07.00.00.mp4
Download successful after 0s (0.0 b, 0.0 bps)
Downloading video for time range 2023-02-19 08:00:00 - 2023-02-19 08:59:59 to /downloads/2023/02/19/G4 Pro (03ee)/G4 Pro (03ee) - 2023-02-19 - 08.00.00.mp4
Download successful after 0s (0.0 b, 0.0 bps)
Downloading video for time range 2023-02-19 09:00:00 - 2023-02-19 09:59:59 to /downloads/2023/02/19/G4 Pro (03ee)/G4 Pro (03ee) - 2023-02-19 - 09.00.00.mp4
Download successful after 0s (0.0 b, 0.0 bps)
3 files downloaded (0.0 b), 0 files skipped, 0 files failed, 3 files total"

This is a huge success for me personally, definitely the farthest I've gotten!  If I put a larger time range where actual files with content are found it fills my drive up but nowhere to be found or... it just times out causing other errors.

Thank you for this tool, this functionality is so desperately needed. (should I merge this with the JHMikkel  discussion?)

**Comment by danielfernau** [comment]
Hi there and thanks for taking the time to ask your question here! No worries, I might have answered one or two times in the past but it's a bit difficult to find these posts :)

To make it easier to understand what the command you're using does:
```plaintext
docker run --volume /downloads:/downloads unifitoolbox/protect-archiver download --address 192.168..** --username .*** --password *** --start 2023-02-18T07:00:00 --end 2023-02-19T12:00:00 /downloads
^--------- ^----------------------------- ^---------------------------- ^------- ^-------------------- ^-------------- ^------------- ^-------------------------- ^------------------------ ^---------
  (1)         (2)                            (3)                          (4)      (5)                   (6)             (7)            (8)                         (9)                       (10)

1  - this tells Docker to run an image
2  - this is our volume mapping, the /downloads on the left side is the directory on your local machine,
     the /downloads on the right side of the : is the directory within the Docker container.
     If you want to download to your user account's download folder, you could use, for example,
     --volume /home/yourusername/Downloads/:/downloads (Linux)
     or
     --volume /Users/yourusername/Downloads/:/downloads (macOS)
3  - this is the name of the image you want Docker to run for you
4  - this is the command with which you tell my tool to download stuff
5  - this is the address of the UniFi Console you want to download from...
6  - ...using this username...
7  - ...and password of a local user on that system...
8  - ...between this date/time...
9  - ...and that date/time...
10 - ...to this directory within the Docker container.
     This last bit relates to the part on the right side of the : in part (2) described above.
     The tool is 'wrapped' in a Docker image/container and this is the output path the tool is instructed to use
     within that container. And the --volume mapping from part (2) is where to map that download directory from
     inside of the container to a folder on your machine.
```

I hope this helps you to understand what's going on and provides you with all the details you need to get it up and running on your end.

Glad you like the tool and happy to provide additional assistance if needed! 🚀 

---

Related discussion: https://github.com/danielfernau/unifi-protect-video-downloader/discussions/147
